nmrfx-analyst-gui/src/main/java/org/nmrfx/processor/gui/spectra/DrawSpectrum.java

```diff
@@ -220,16 +176,9 @@ public static float[] getLevels(DatasetAttributes fileData) {
         return levels;
     }
 
-    AXMODE[] getAxModes() {
-        AXMODE[] modes = new AXMODE[axModes.length];
-        System.arraycopy(axModes, 0, modes, 0, modes.length);
-        return modes;
-    }
-
-    Axis[] getAxes() {
-        Axis[] tempAxes = new Axis[axes.length];
-        System.arraycopy(axes, 0, tempAxes, 0, axes.length);
-        return tempAxes;
+    PolyChartAxes getAxes() {
+        //XXX previous implementation was making a copy of the backing array: is that necessary due to multithreading?
```

Comment:
@brucejohnson here's a question for you. If necessary, I can add a way to work on a copy as well this way.